sql/wsrep_client_service.cc

```diff
@@ -304,6 +304,13 @@ enum wsrep::provider::status Wsrep_client_service::replay()
     replayer_service.replay_status(ret);
   }
 
+  // Original thd binlog stmt cache could contain
+  // changes on non-transactional objects like sequences
+  // In Galera we allow only InnoDB sequences, thus
+  // sequence table updates are in writeset.
+  // Binlog cache needs reset so that binlog_close
```

Comment:
"Binlog cache needs reset so that binlog_close" 
This sentence feels like it was truncated.

Reply:
missing part : does not write binlog cache to binlog file

Reply:
I still don't understand why this change is needed. 
Commenting it out won't cause failures in the tests that come with this patch.

Reply:
It is needed, if you remove this line you will have following kind of assertion:

mariadbd: /home/jan/work/mariadb/10.6/sql/log.cc:1765: int binlog_close_connection(handlerton*, THD*): Assertion `cache_mngr->stmt_cache.empty()' failed.

in test case

galera.galera_sequences_transaction 'binlogon' [ fail ]

This does not happen every run.